m2cgen/assemblers/boosting.py

```diff
@@ -134,12 +139,30 @@ def _assemble_child_tree(self, tree, child_id):
 class LightGBMModelAssembler(BaseBoostingAssembler):
 
     classifier_name = "LGBMClassifier"
-
+    
     def __init__(self, model):
-        model_dump = model.booster_.dump_model()
-        trees = [m["tree_structure"] for m in model_dump["tree_info"]]
+        if isinstance(model, lightgbm.Booster):
+            model_dump = model.dump_model()
+        else:
+            model_dump = model.booster_.dump_model()
 
-        super().__init__(model, trees)
+        trees = [m["tree_structure"] for m in model_dump["tree_info"]]
+ 
+        is_classification = False
+        output_size = 1
+        if 'num_class' in model_dump:
+            output_size = model_dump['num_class']
+            
+            if 'objective' not in model_dump:
+                raise Exception("objective function not specified in model_dump. try upgrading lightgbm >= 2.3.0")
+
+            objective = model_dump['objective']
+            is_classification = 'multiclass' in objective or 'binary' in objective
+            
+            if output_size > 2:
+                output_size = output_size
+
+        super().__init__(model, trees, output_size=output_size, is_classification=is_classification)
```

Comment:
I wonder how this code is different from the following one:
```python
    if isinstance(model, lightgbm.Booster):
        model_dump = model.dump_model()
    else:
        model_dump = model.booster_.dump_model()
    trees = [m["tree_structure"] for m in model_dump["tree_info"]]
    super().__init__(model, trees)
```
Given that the base constructor remains intact?

Reply:
I believe this is not enough since `type(model).__name__` is no longer a reliable source of truth about whether this is a classifier or regressor.

Reply:
Ahhh... got it. So now `model` variable which pass to base constructor sometimes will be of type `LGBMRegressor`/`LGBMClassifier` and sometimes of type `Booster`. This makes me think something is fundamentally wrong here. 

Although one possible workaround could be start treating `trees` itself as our model. Any thoughts?

Reply:
I don't think `model` ever ends up being used, I'd prefer to remove it entirely. 

If we pass trees around I'm not sure how much reuse we'd get as LightGBM and XGBoost have different properties